test/ExecutionFramework.cpp

```diff
@@ -40,27 +42,47 @@ using namespace solidity::util;
 using namespace solidity::test;
 
 ExecutionFramework::ExecutionFramework():
-	ExecutionFramework(solidity::test::CommonOptions::get().evmVersion())
+	ExecutionFramework(solidity::test::CommonOptions::get().evmVersion(), solidity::test::CommonOptions::get().vmPaths)
 {
 }
 
-ExecutionFramework::ExecutionFramework(langutil::EVMVersion _evmVersion):
+ExecutionFramework::ExecutionFramework(langutil::EVMVersion _evmVersion, std::vector<boost::filesystem::path> const& _vmPaths):
 	m_evmVersion(_evmVersion),
 	m_optimiserSettings(solidity::frontend::OptimiserSettings::minimal()),
 	m_showMessages(solidity::test::CommonOptions::get().showMessages),
-	m_evmHost(make_shared<EVMHost>(m_evmVersion))
+	m_vmPaths(_vmPaths)
 {
 	if (solidity::test::CommonOptions::get().optimize)
 		m_optimiserSettings = solidity::frontend::OptimiserSettings::standard();
 
+	for (auto const& path: m_vmPaths)
+		if (EVMHost::getVM(path.string()).has_capability(EVMC_CAPABILITY_EWASM))
+			m_supportsEwasm = true;
+
+	setEvm(evmc_capabilities::EVMC_CAPABILITY_EVM1);
+}
+
+void ExecutionFramework::setEvm(evmc_capabilities _cap)
+{
+	m_evmcHost.reset();
+	for (auto const& path: m_vmPaths)
+	{
+		evmc::VM& vm = EVMHost::getVM(path.string());
+		if (vm.has_capability(_cap))
+			m_evmcHost = std::unique_ptr<EVMHost, std::function<void(EVMHost*)>>(
```

Comment:
Why the custom deleter?

Reply:
Also maybe add a `break`?

Reply:
In the header `EVMHost` is just forward declared. It is not possible to use the default-deleter on an incomplete type.

Reply:
Ok sure, but why don't you properly declare it?

Reply:
If it doesn't work without custom deleter then please use a shared_ptr instead of using a custom deleter.

Reply:
Ah nice, it can be just properly declared. I was thinking that it would not work, because it was forward declared. Next time I will just try ;)